packages/editor/src/store/utils/notice-builder.js

```diff
@@ -97,14 +97,18 @@ export function getNotificationArgumentsForSaveFail( data ) {
 	// If the post was being published, we show the corresponding publish error message
 	// Unless we publish an "updating failed" message
 	const messages = {
-		publish: __( 'Publishing failed' ),
-		private: __( 'Publishing failed' ),
-		future: __( 'Scheduling failed' ),
+		publish: __( 'Publishing failed.' ),
+		private: __( 'Publishing failed.' ),
+		future: __( 'Scheduling failed.' ),
 	};
-	const noticeMessage = ! isPublished && publishStatus.indexOf( edits.status ) !== -1 ?
+	let noticeMessage = ! isPublished && publishStatus.indexOf( edits.status ) !== -1 ?
 		messages[ edits.status ] :
-		__( 'Updating failed' );
+		__( 'Updating failed.' );
 
+	// Check if message string contains html tags. It is unsafe to inject unfiltered html here.
```

Comment:
Technically it's not a matter of safety, at least not without other changes to inject the HTML as `innerHTML`. Even if there were HTML included in the notice message, it would be written as plaintext (for example, `"Publishing failed. Error message: <strong>Bad things</strong> happened."`). It's still not ideal from a user's perspective to have shown (at least not without stripping the tags), but not a security concern.

Reply:
Do you want me to change the comment or strip the tags?

Reply:
> Do you want me to change the comment or strip the tags?

Per your comment at https://github.com/WordPress/gutenberg/pull/15657#discussion_r285184131 (which I tend to agree with), stripping the tags might yield some bizarre results (at least for specific types of elements; `em`, `strong`, and others may be fine).

Changing the comment to clarify this could improve. Maybe:

```suggestion
	// Check if message string contains HTML. Notice text is currently only
	// supported as plaintext, and stripping the tags may muddle the meaning.
```

Reply:
Merge your change. 👍